src/main/resources/reference.conf

```diff
@@ -1290,20 +1290,45 @@ resourceTypes = {
       "unlock_resource" = {
         description = "Can unlock a resource"
       }
+      "read_aggregate_data" = {
+        description = "Can read aggregate data about a snapshot"
+      }
+      "create_snapshot_request" = {
+        description = "Can create a snapshot request for this snapshot"
+      }
+      "get_snapshot_builder_settings" = {
+        description = "Can get the snapshot builder settings for this snapshot"
+      }
+      "update_snapshot_builder_settings" = {
+        description = "Can get the snapshot builder settings for this snapshot"
+      }
     }
     ownerRoleName = "steward"
     roles = {
```

Comment:
I'd like to suggest a reordering of role definitions as follows, which I think will underscore the flow of role inclusion:

```
steward (includes custodian roles)
custodian (includes reader roles)
reader (includes aggregate_data_reader roles)
aggregate_data_reader (includes discoverer roles)
discoverer
snapshot_builder_manager
admin
```